src/main/java/org/opentripplanner/routing/algorithm/filterchain/deletionflagger/FlexOnlyToDestinationFilter.java

```diff
@@ -0,0 +1,42 @@
+package org.opentripplanner.routing.algorithm.filterchain.deletionflagger;
+
+import java.time.Duration;
+import java.util.List;
+import org.opentripplanner.model.plan.Itinerary;
+import org.opentripplanner.model.plan.Leg;
+import org.opentripplanner.routing.algorithm.filterchain.ItineraryListFilter;
+
+/**
+ * Remove all itineraries where all the final leg is more than x minutes of walking.
+ */
+public class FlexOnlyToDestinationFilter implements ItineraryListFilter {
```

Comment:
Would it make sense to implement this similar to how `maxTransferDuration` is used, that way we could do the filtering already before the search?

Reply:
Wouldn't that lead to results which have longer cycling instead? I guess this isn't what we want, or is it?

Reply:
That's actually a very good point which I had not thought about.